src/App.ts

```diff
@@ -470,9 +471,16 @@ export default class App {
 
   // TODO: just make a type alias for Middleware<SlackEventMiddlewareArgs<'message'>>
   // TODO: maybe remove the first two overloads
-  public message(...listeners: Middleware<SlackEventMiddlewareArgs<'message'>>[]): void;
-  public message(pattern: string | RegExp, ...listeners: Middleware<SlackEventMiddlewareArgs<'message'>>[]): void;
-  public message(...patternsOrMiddleware: (string | RegExp | Middleware<SlackEventMiddlewareArgs<'message'>>)[]): void {
+  public message<Subtype extends string | undefined = '*'>(
```

Comment:
I don't think this asterisk string is cool but what else can be relevant here?

Reply:
Asterisk made sense to me while reading it over. 👍🏻